src/compiler/utilities.ts

```diff
@@ -5360,20 +5360,16 @@ namespace ts {
         return getStringFromExpandedCharCodes(expandedCharCodes);
     }
 
+    export function readJsonOrUndefined(path: string, hostOrText: { readFile(fileName: string): string | undefined } | string): object | undefined {
+        const jsonText = isString(hostOrText) ? hostOrText : hostOrText.readFile(path);
+        if (!jsonText) return undefined;

```

Comment:
Is this going to return `undefined` when `jsonText === ""` and, if so, is that desirable?

Reply:
We expect object literal os "" is going to error later anyways so this is desirable